uaclient/clouds/gcp.py

```diff
@@ -42,11 +42,40 @@ def __init__(self):
     # mypy does not handle @property around inner decorators
     # https://github.com/python/mypy/issues/1362
     @property  # type: ignore
-    @util.retry(HTTPError, retry_sleeps=[1, 2, 5])
+    @util.retry(exceptions.GCPProAccountError, retry_sleeps=[1, 2, 5])
     def identity_doc(self) -> Dict[str, Any]:
-        url_response, _headers = util.readurl(
-            TOKEN_URL, headers={"Metadata-Flavor": "Google"}
-        )
+        try:
+            headers = {"Metadata-Flavor": "Google"}
+            url_response, _headers = util.readurl(TOKEN_URL, headers=headers)
+        except URLError as e:
+            body = None
+            if hasattr(e, "body"):
+                body = getattr(e, "body", None)
+            elif hasattr(e, "read"):
+                reader = getattr(e, "read")()
+                body = reader.decode("utf-8")
```

Comment:
We can simplify this by only catching `HTTPError`s which is what happened in the initial bug report.
```suggestion
        except HTTPError as e:
            body = getattr(e, "body", None)
```

Reply:
@CalvoM we can still simplify what is there to this suggestion :)